### PR TITLE
fix(List): add *add Button* to the toolbar

### PR DESCRIPTION
**Pull request by jmfrancois**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

You can't add a new Item to the list, so it's not full features

**What is the new behavior?**

If you add *onClickAdd* props to the list, you will see an Add button which will be triggered onClick.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
screen shot 2016-10-24 at 12 01 06

### CODE REVIEW

**Comment by acateland** [comment]
maybe add its not the only action wich is missing from the toolbar and we should approach a more genereic solution ?
like have the ability to pass an optional props describing a list of action that can be added to the bar 
```
[{
  eventsHandlers,
  id,
  label,
  icon,
},
{
  eventsHandlers,
  id,
  label,
  icon,
},]
```
and make the mechanics reusable ?

**Comment by jsomsanith** [comment]
I agree with @acateland. Each project can have their specific actions too.
For now in TDP, we have a "create folder", I don't know if this button will stay here but the proposed solution will be flexible enough to add/remove actions in toolbar and avoid to change the component's code

**Comment by jmfrancois** [comment]
actions support added. Please @jsomsanith could you review it again  ?

File: src/List/Toolbar/Toolbar.component.js [hunk 1]
```diff
@@ -23,8 +25,39 @@ function Toolbar(props) {
...
+	if (props.onClickAdd) {
+		add = {
+			label: 'Add',
+			icon: 'fa fa-plus',
+			onClick: props.onClickAdd,
+		};
+	}
 	return (
 		<Navbar componentClass="div" role="toolbar" fluid>
+			{add ? (
```
Comment (jsomsanith) [inline src/List/Toolbar/Toolbar.component.js]: Is it ok to test add to display add + listActions ?

File: src/List/Toolbar/Toolbar.component.js [hunk 2]
```diff
@@ -23,8 +25,39 @@ function Toolbar(props) {
...
+				<Nav>
+					<Action className="navbar-btn" bsStyle="success" {...add} />
+					<ButtonGroup className="navbar-btn">
+						{props.listActions ? (
+							<DropdownButton title="" id="tc-list-toolbar-add-dropdown" bsStyle="link">
```
Comment (jsomsanith) [inline src/List/Toolbar/Toolbar.component.js]: empty title

File: src/List/Toolbar/Toolbar.test.js [hunk 3]
```diff
@@ -33,4 +33,18 @@ describe('Toolbar', () => {
 		expect(subProps.sortBy).toBe(props.sortBy);
 		expect(Object.keys(subProps).length).toBe(2);
 	});
+	it('should support listActions', () => {
```
Comment (jsomsanith) [inline src/List/Toolbar/Toolbar.test.js]: Missing behavior test (click on add button and on action list item)

File: src/List/Toolbar/Toolbar.component.js [hunk 4]
```diff
@@ -23,8 +25,39 @@ function Toolbar(props) {
...
+									const onClick = e => action.onClick(e, action);
+									return (
+										<MenuItem
+											onClick={onClick}
+											eventKey={index}
+										>
+											<Icon name={action.icon} />
+											{action.label}
+										</MenuItem>
+									)
```
Comment (jsomsanith) [inline src/List/Toolbar/Toolbar.component.js]: Missing semi-colon

File: src/List/DisplayTable/DisplayTable.component.js [hunk 5]
```diff
@@ -6,38 +6,38 @@ import Actions from '../../Actions';
...
 function RowRenderer(props) {
-	const keys = props.columns.map(column => column.key); // {key, label}
-	const row = [];
-	keys.forEach(key => row.push(props.row[key]));
 	return (
 		<tr>
-			{row.map((column, index) => {
-				if (props.columns[index].key === props.titleKey) {
-					const onClick = event => props.onTitleClick(props.row, event);
+			{props.columns.map((column, index) => {
```
Comment (frassinier) [inline src/List/DisplayTable/DisplayTable.component.js]: Perhaps, `columns` should be mandatory?

Comment (acateland) [inline src/List/DisplayTable/DisplayTable.component.js]: @frassinier corrected check your review please

Comment (frassinier) [inline src/List/DisplayTable/DisplayTable.component.js]: It was just a comment but GJ :ok_hand:

**Pull request merged**